src/UI/Component/ViewControl/Factory.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\ViewControl;
+/**
+ * This is how the factory for UI elements looks.
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      Mode View Controls enable the switching between different aspects of some data. The different modes are mutually
+	 *      exclusive and can therefore not be activated at once.
+	 *   composition: >
+	 *      Mode View Controls are composed of Buttons switching between active and inactive states.
+	 *   effect: >
+	 *      Clicking on an inactive Button turns this button active and all other inactive. Clicking on an active button
+	 *      has no effect.
+	 *
+	 * rules:
+	 *   usage:
+	 *      1: Exactly one Button MUST always be active.
+	 *   accessibility:
+	 *      1: The Buttons of the Mode View Control MUST cary the role-attribute "group".
+	 *      2: The Buttons of the Mode View Control MUST set an aria-label clearly describing it's functionality.
```

Comment:
Most of the UI components are going to use this assistive attributes. Can I add "aria-label" and "aria-checked" attributes in the Standard button? or a whole new button has to be implemented? 
Active button example:
$tpl->setVariable("BUTTON", $default_renderer->render($f->button()->standard($label, $action,$aria_label,$aria_checked)->withUnavailableAction()));